src/m365/viva/commands/engage/engage-community-add.ts

```diff
@@ -0,0 +1,220 @@
+      { option: '--adminEntraUserNames [adminEntraUserNames]' },
+      { option: '--wait' }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.displayName && args.options.displayName.length > 255) {
+          return `The maximum amount of characters for 'displayName' is 255.`;
+        }
+
+        if (args.options.description && args.options.description.length > 1024) {
+          return `The maximum amount of characters for 'description' is 1024.`;
+        }
+
+        if (args.options.privacy && this.privacyOptions.indexOf(args.options.privacy) === -1) {
+          return `'${args.options.privacy}' is not a valid value for privacy. Allowed values are: ${this.privacyOptions.join(', ')}.`;
+        }
+
+        if (args.options.adminEntraIds && args.options.adminEntraUserNames) {
+          return `You can only specify either 'adminEntraIds' or 'adminEntraUserNames'`;
+        }
+
+        if (args.options.adminEntraIds) {
+          const adminEntraIds = args.options.adminEntraIds.split(',');
+          for (let i: number = 0; i < adminEntraIds.length; i++) {
+            const trimmedId: string = adminEntraIds[i].trim();
+            if (!validation.isValidGuid(trimmedId)) {
+              return `${trimmedId} is not a valid GUID`;
+            }
+          }
+        }
+
+        if (args.options.adminEntraUserNames) {
+          const adminEntraUserNames = args.options.adminEntraUserNames.split(',');
+          for (let i: number = 0; i < adminEntraUserNames.length; i++) {
+            const trimmedUPN: string = adminEntraUserNames[i].trim();
+            if (!validation.isValidUserPrincipalName(trimmedUPN)) {
+              return `${trimmedUPN} is not a valid UPN`;
+            }
+          }
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initTypes(): void {
+    this.types.string.push('displayName', 'description', 'privacy', 'adminEntraIds', 'adminEntraUserNames');
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    const { displayName, description, privacy, adminEntraIds, adminEntraUserNames, wait } = args.options;
+
+    if (this.verbose) {
+      logger.logToStderr(`Creating a Viva Engage community with display name '${displayName}'...`);
+    }
+
+    try {
+      const requestOptions: CliRequestOptions = {
+        url: `${this.resource}/beta/employeeExperience/communities`,
+        headers: {
+          accept: 'application/json;odata.metadata=none',
+          'content-type': 'application/json;odata=nometadata'
+        },
+        responseType: 'json',
+        fullResponse: true,
+        data: {
+          displayName: displayName,
+          description: description,
+          privacy: privacy
+        }
+      };
+
+      let entraIds: string[] = [];
+
+      if (adminEntraIds) {
+        entraIds = adminEntraIds.split(',').map(id => this.getUserIdUrl(id));
```

Comment:
To split a comma-separated string, we have a util in the `formatting` util for this.